src/network-validation/network-validation.service.ts

```diff
@@ -0,0 +1,84 @@
+import { MikroORM, UseRequestContext } from '@mikro-orm/core';
+import { Inject, Injectable, LoggerService } from '@nestjs/common';
+import { LidoLocator, LIDO_LOCATOR_CONTRACT_TOKEN } from '@lido-nestjs/contracts';
+import { LOGGER_PROVIDER } from '@lido-nestjs/logger';
+import { ConfigService } from '../common/config';
+import { ConsensusProviderService } from '../common/consensus-provider';
+import { ExecutionProviderService } from '../common/execution-provider';
+import { RegistryKeyStorageService, RegistryOperatorStorageService } from '../common/registry';
+import { SRModuleStorageService } from '../storage/sr-module.storage';
+import { AppInfoStorageService } from '../storage/app-info.storage';
+import { MODULE_ADDRESSES_FOR_CHAINS } from './network-validation.constants';
+
+@Injectable()
+export class NetworkValidationService {
+  constructor(
+    protected readonly orm: MikroORM,
+    @Inject(LIDO_LOCATOR_CONTRACT_TOKEN) protected readonly contract: LidoLocator,
+    @Inject(LOGGER_PROVIDER) protected readonly logger: LoggerService,
+    protected readonly configService: ConfigService,
+    protected readonly consensusProviderService: ConsensusProviderService,
+    protected readonly executionProviderService: ExecutionProviderService,
+    protected readonly keyStorageService: RegistryKeyStorageService,
+    protected readonly moduleStorageService: SRModuleStorageService,
+    protected readonly operatorStorageService: RegistryOperatorStorageService,
+    protected readonly appInfoStorageService: AppInfoStorageService,
+  ) {}
+
+  @UseRequestContext()
+  public async validate(): Promise<void> {
```

Comment:
let's break the method into smaller methods

Reply:
I moved the logic of checking that chin ID from the env variables, EL chain ID, and CL chain ID match each other to the separate private method. Despite this change, I feel that it doesn't make much sense to split the logic of the `validate` method into smaller methods and functions. I think that the logic of this method is holistic and it would be harder to understand it if it is artificially split into separate methods and functions.
If you have any particular ideas about what parts of this code should be extracted to a separate method, let's discuss.